portfolio/src/main/webapp/script.js

```diff
@@ -55,21 +55,49 @@ function pauseProjectChange() {
   clearTimeout(changeCurrProj);
 }
 
-/** Fetches comment(s) and updates the UI to display */
-function fetchCommentAndDisplay() {
+/** Fetches comment(s) and updates the UI to display them */
+function fetchCommentAndDisplay(numOfCommentsToDisplay) {
   fetch('/data').then(response => response.json()).then((comments) => {
-    const commentsContainer = document.getElementById('comments-container');
-    comments.forEach((comment) => {
-      const currentCommentContainer = createDivElement();
-      currentCommentContainer.style.padding = "50px 0px";
-      commentsContainer.appendChild(currentCommentContainer);
-      currentCommentContainer.appendChild(createParagraphElement(comment.name));
-      currentCommentContainer.appendChild(createParagraphElement(comment.email));
-      currentCommentContainer.appendChild(createParagraphElement(comment.text));
-    });
+    /** 
+    * Error check to ensure the user did not select to display more comments than exist in the datastore
+    *
+    * UI is not updated on user selecting to view more comments than exist in the datastore
+    */ 
+    if (numOfCommentsToDisplay > comments.length) {
+      alert("The selected value is greater than the total number of comments.");
+    }
+    else {
+
+      // Clear out old comments before inserting new comments into the DOM
+      const commentsContainer = document.getElementById('comments-container');
+      commentsContainer.innerHTML = "";
+        
+      // Display selected number of comments on the page
+      var i = 0;
+      while (i < numOfCommentsToDisplay) {
+        let currentComment = comments[i];
+        const currentCommentContainer = createDivElement();
+        currentCommentContainer.style.padding = "50px 0px";
+        commentsContainer.appendChild(currentCommentContainer);
+        currentCommentContainer.appendChild(createParagraphElement(currentComment.name));
+        currentCommentContainer.appendChild(createParagraphElement(currentComment.email));
+        currentCommentContainer.appendChild(createParagraphElement(currentComment.text));
+        i++;
+      }
+    }
   });
 }
 
+function getNumCommentsSelectedAndDisplay() {
+  
+  // Retrieve number of comments selected by the user
+  const selectElement = document.getElementById('num-comments');
+  const numOfCommentsToDisplay = selectElement.options[selectElement.selectedIndex].value;
+
+  // Call to update UI
+  fetchCommentAndDisplay(numOfCommentsToDisplay);
```

Comment:
since numCommentsToDisplay is user input, you'll have to be careful that it's numeric and within an expected range. A user could modify the selectElement with inspect element to pass whatever value they want.

Reply:
is it ok to add these checks in the DataServlet?

Reply:
Definitely. Server code should always validate input from the client to defend against malicious requests.